### PR TITLE
Add comment and vote level for forum posts

### PR DESCRIPTION
**Pull request by motechFR**
### <samp>🤖 Generated by Copilot at 606c6a7</samp>

Added a new permission level `vote_comment` for forum members to allow them to vote on comments in different post categories. Updated the database schema, the Prisma schema, and the forum permission components to support this feature.

### <samp>🤖 Generated by Copilot at 606c6a7</samp>

*  Add `vote_comment` as a new post category permission level that allows voting and commenting on posts ([link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-8a01b2705a2ace54eeac124ce76e360ff43ca95bb00e23b104ff8fd9d6c03cb0R7), [link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-7b834cea8b8e48599e19d399d68d6b77113a44fc9f99786b380e80d2875f9779R9), [link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-c7697e855acb9f89caad5c051e0d98a27eff95f48b05c6e921c26e550f5b1746R1-R2), [link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-5b443964f4f3a611682db8f7e02177b0a8c632b2039e2bd5e4dd7347815c565cR1349))
*  Update the forum member permission mapping to include `vote_comment` as an inherited permission ([link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-7b834cea8b8e48599e19d399d68d6b77113a44fc9f99786b380e80d2875f9779R18))
*  Import `forumMemberPermissionOptions` from `shared` module and use its labels for the forum member permissions in `PostCategoryPermissionRow` component ([link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-6033591e0274a77e66f53d6d4a689b9c91973aed01b06ed3c34e0c18133f0ddcL15-R15), [link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-6033591e0274a77e66f53d6d4a689b9c91973aed01b06ed3c34e0c18133f0ddcL47-R48))
*  Remove an empty line from the `shared` module for formatting purposes ([link](https://github.com/charmverse/app.charmverse.io/pull/2032/files?diff=unified&w=0#diff-8a01b2705a2ace54eeac124ce76e360ff43ca95bb00e23b104ff8fd9d6c03cb0L15))

### CODE REVIEW

File: components/forum/components/permissions/PostCategoryPermissionRow.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { useRoles } from 'hooks/useRoles';
 import type { PostCategoryPermissionInput } from 'lib/permissions/forum/upsertPostCategoryPermission';
 import type { TargetPermissionGroup } from 'lib/permissions/interfaces';
 
-import { postCategoryPermissionLabels } from './shared';
+import { forumMemberPermissionOptions, postCategoryPermissionLabels } from './shared';
```
Comment (mattcasey) [inline components/forum/components/permissions/PostCategoryPermissionRow.tsx:15]: i forgot to delete forumMemberPermissionOptions - this isn't shared by anything else, and i found it easier to know what's going on by having the options defined here, locally

Comment (motechFR) [reply components/forum/components/permissions/PostCategoryPermissionRow.tsx:15]: It's also being used in this component
components/forum/components/permissions/PostCategoryPermissionAddRolesDialog.tsx

I re-added it here, so we have a single source of truth on the options

File: components/forum/components/permissions/shared.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export const postCategoryPermissionLabels: Record<PostCategoryPermissionLevel, s
   category_admin: 'Category Admin',
   moderator: 'Moderator',
   full_access: 'Post & Comment',
```
Comment (mattcasey) [inline components/forum/components/permissions/shared.ts:6]: Full access should now say "Post, Vote & Comment"

File: prisma/schema.prisma [hunk 3]
```diff
@@ -1346,6 +1346,7 @@ enum PostCategoryPermissionLevel {
   category_admin
   moderator
   full_access
+  vote_comment
```
Comment (mattcasey) [inline prisma/schema.prisma:1349]: suggestion: "add_comment_vote" instead, to closer align with what we did for proposals, or at least put comment first

**Review by mattcasey — APPROVED** [review]
Approved but had some suggestions. Also, in this PR or a separate one, can you please make the post and proposal category labels consistent?

**Pull request merged**